src/Raven.Server/ServerWide/ServerStore.cs

```diff
@@ -1473,6 +1474,63 @@ private static bool DatabaseNeedsToRunIdleOperations(DocumentDatabase database)
             return ((now - maxLastWork).TotalMinutes > 5) || ((now - database.LastIdleTime).TotalMinutes > 10);
         }
 
+        public void AssignNodesToDatabase(ClusterTopology clusterTopology, DatabaseRecord record)
+        {
+            var topology = record.Topology;
+
+            Debug.Assert(topology != null);
+
+            var clusterNodes = clusterTopology.Members.Keys
+                .Concat(clusterTopology.Promotables.Keys)
```

Comment:
We shouldn't add it to promotables. They may take a while to come up and actually figure that one out.
Only members / watchers should be there.

Reply:
done